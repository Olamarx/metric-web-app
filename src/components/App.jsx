import React from 'react';
import { useSelector } from 'react-redux';
import { Routes, Route, Navigate } from 'react-router-dom';
import unidecode from 'unidecode';
import Continents from './Continents/Continents';
import Country from './Country/Country';

function App() {
  const contr = useSelector((state) => state.country);
  const path = `/continents/country/:${unidecode(contr.name.replace(/ /gi, '_'))}`;
  return (
    <div>

      <Routes>
        <Route path="/" element={<Navigate to="/continents" />} />
        <Route path="/continents/*" element={<Continents />} />
        <Route path={path} element={<Country />} />
      </Routes>
    </div>
  );
}

export default App;
