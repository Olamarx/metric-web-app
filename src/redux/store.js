import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './Countries';
// import countryDetails, { loadCountry } from './MoreDetails';

const store = configureStore({
  reducer: {
    countriesReducer,
  },
});

export default store;
