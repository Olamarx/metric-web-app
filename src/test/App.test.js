import renderer, { act } from 'react-test-renderer';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import App from '../components/App';

describe('Body of the whole landing page', () => {
  it('checks for the name of the country', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Provider store={store}>
            <App />
          </Provider>
        </MemoryRouter>,
      );
    });
  });
  waitFor(() => {
    const chillForDOM = screen.findByText(/Namibia/);
    expect(chillForDOM).not.toBeNull();
  });

  it('checks for the name of the country', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Provider store={store}>
            <App />
          </Provider>
        </MemoryRouter>,
      );
    });
    waitFor(() => {
      const chillForDOM = screen.findByText(/Eswatini/);
      expect(chillForDOM).not.toBeNull();
    });
  });

  it('checks for the name of the country', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Provider store={store}>
            <App />
          </Provider>
        </MemoryRouter>,
      );
    });
    waitFor(() => {
      const chillForDOM = screen.findByText(/Sudan/);
      expect(chillForDOM).not.toBeNull();
    });
  });

  it('checks for the name of the country', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Provider store={store}>
            <App />
          </Provider>
        </MemoryRouter>,
      );
    });
    waitFor(() => {
      const chillForDOM = screen.findByText(/British Indian Ocean Territory/);
      expect(chillForDOM).not.toBeNull();
    });
  });

  it('checks for the name of the country', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Provider store={store}>
            <App />
          </Provider>
        </MemoryRouter>,
      );
    });
    waitFor(() => {
      const chillForDOM = screen.findByText(/Burkina Faso/);
      expect(chillForDOM).not.toBeNull();
    });
  });

  it('checks for the name of the country', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Provider store={store}>
            <App />
          </Provider>
        </MemoryRouter>,
      );
    });
    waitFor(() => {
      const chillForDOM = screen.findByText(/Rwanda/);
      expect(chillForDOM).not.toBeNull();
    });
  });

  it('matching snapshot', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
