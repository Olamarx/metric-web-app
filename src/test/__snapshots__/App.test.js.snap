// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body of the whole landing page matching snapshot 1`] = `<div />`;

exports[`matching snapshot 1`] = `<div />`;
